### PR TITLE
Fix profile pic banner pic upload

### PR DESCRIPTION
**Pull request by scoder1747**
## Summary
The existing picture-upload flow for Profile and Banner is broken. Root cause analysis of this bug is due to missing KeyPair parameter being passed while instantiating EditPictureControl objects. They KeyPair param is used to create nip98 signature and which is eventually used to create an "Authorization" header for creating an upload request.

## Checklist

- [ ] I have read (or I am familiar with) the [Contribution Guidelines](../docs/CONTRIBUTING.md)
- [x] I have tested the changes in this PR
- [ ] My PR is either small, or I have split it into smaller logical commits that are easier to review
- [ ] I have added the signoff line to all my commits. See [Signing off your work](../docs/CONTRIBUTING.md#sign-your-work---the-developers-certificate-of-origin)
- [ ] I have added appropriate changelog entries for the changes in this PR. See [Adding changelog entries](../docs/CONTRIBUTING.md#add-changelog-changed-changelog-fixed-etc)
    - [ ] I do not need to add a changelog entry. Reason: _[Please provide a reason]_
- [ ] I have added appropriate `Closes:` or `Fixes:` tags in the commit messages wherever applicable, or made sure those are not needed. See [Submitting patches](https://github.com/damus-io/damus/blob/master/docs/CONTRIBUTING.md#submitting-patches)

### CODE REVIEW

**Comment by jb55** [comment]
This looks right to me!

**Comment by jb55** [comment]
although nostr.build supports uploads without auth right? was it not working because the image was too large?

**Review by jb55 — APPROVED** [review]

**Comment by scoder1747** [comment]
> although nostr.build supports uploads without auth right? was it not working because the image was too large?

I thought it would support but as of now, the following is the response sent by nostr-build
Upload Error: Unauthorized, please provide a valid nip-98 token

There is one more commit that limits the profile and banner upload to image only and sets selection quantity to 1

File: damus/Views/MediaPicker.swift [hunk 1]
```diff
@@ -115,9 +119,14 @@ struct MediaPicker: UIViewControllerRepresentable {
...
-        configuration.filter = imagesOnly ? .images : .any(of: [.images, .videos])
-        configuration.selection = .ordered // images are returned in the order they were selected + numbered badge displayed
+        if case .postView = mediaPickerEntry {
+            configuration.selectionLimit = 0 // Allows multiple media selection
+            configuration.filter = .any(of: [.images, .videos])
+            configuration.selection = .ordered // images are returned in the order they were selected + numbered badge displayed
+        } else if case .editPictureControl = mediaPickerEntry {
+            configuration.selectionLimit = 1 // Allows one media selection
+            configuration.filter = .images
+        }
```
Comment (jb55) [inline damus/Views/MediaPicker.swift:129]: please use switch statement

**Review by jb55 — CHANGES_REQUESTED** [review]

File: damus/Views/MediaPicker.swift (hunk 1 above)
Comment (scoder1747) [reply damus/Views/MediaPicker.swift]: @jb55 I have made the update

**Pull request closed without merging**